components/ai_chat/core/browser/leo_local_models_updater.h

```diff
@@ -0,0 +1,75 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_AI_CHAT_CORE_BROWSER_LEO_LOCAL_MODELS_UPDATER_H_
+#define BRAVE_COMPONENTS_AI_CHAT_CORE_BROWSER_LEO_LOCAL_MODELS_UPDATER_H_
+
+#include <string>
+
+#include "base/files/file_path.h"
+#include "base/observer_list.h"
+#include "base/observer_list_types.h"
+#include "brave/components/brave_component_updater/browser/brave_component.h"
+
+class LeoLocalModelsUpdaterTest;
+
+namespace ai_chat {
+
+extern const char kUniversalQAModelName[];
+
+// A component updater for downloading local models for Leo.
+class LeoLocalModelsUpdater : public brave_component_updater::BraveComponent {
+ public:
+  class Observer : public base::CheckedObserver {
+   public:
+    virtual void OnLeoLocalModelsReady() {}
+
+   protected:
+    ~Observer() override = default;
+  };
+
+  LeoLocalModelsUpdater(BraveComponent::Delegate* component_delegate,
+                        const base::FilePath& user_data_dir);
+  LeoLocalModelsUpdater(const LeoLocalModelsUpdater&) = delete;
+  LeoLocalModelsUpdater& operator=(const LeoLocalModelsUpdater&) = delete;
+  ~LeoLocalModelsUpdater() override;
+
+  virtual void Register();
+  // Method for deleting component updater directory. Path is relative to
+  // user_data_dir passed in the constructor. If the user_data_dir is empty, it
+  // won't have any effect.
+  void Cleanup(base::OnceCallback<void(bool)> reply_callback = {});
+
+  virtual const base::FilePath& GetUniversalQAModel() const;
+
+  void AddObserver(Observer* observer);
+  void RemoveObserver(Observer* observer);
+
+ protected:
+  void OnComponentReady(const std::string& component_id,
+                        const base::FilePath& install_dir,
+                        const std::string& manifest) override;
+
+ private:
+  friend class ::LeoLocalModelsUpdaterTest;
+
+  static void SetComponentIdAndBase64PublicKeyForTest(
+      const std::string& component_id,
+      const std::string& component_base64_public_key);
+  static void SetUserDataDirForTest(const base::FilePath& user_data_dir);
+  static std::string g_component_id_;
+  static std::string g_component_base64_public_key_;
+  static base::FilePath g_user_data_dir_for_test_;
```

Comment:
please do not use non-POD statics. Always go for lazy construction like so:
```
std::string& GetComponentId() {
  static base::NoDestructor<std::string> component_id{...};
  return *component_id;
}
```

https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables

Reply:
We no longer need that with 7e443e164c9e4952e57e1d040e9f300d7fa521e2.
I will keep that in mind next time, thanks.